CDCgov/prime-reportstream
# change quality filter to block the negative Antigen Test results

This PR address quality filter to filter out Negative Antigen test result. 

Test Steps:
1. Clone the PR
2. ./gradlew clean package
3. Add ca-phd setting and prescryptive as below to organization.yml 
   - name: prescryptive
     description: Prescryptive Health
     jurisdiction: FEDERAL
     senders:
       - name: default
         organizationName: prescryptive
         topic: covid-19
         customerStatus: active
         schemaName: hl7/prescryptive-covid-19
         format: HL7
4. run docker-compose up
5. ./gradlew reloadSettings
6. Download the Negative and Positive Antigen Test message files to ./junk/CA
7. Run curl -X POST -H "client:prescryptive" -H "content-type:application/hl7-v2" --data-binary "@./junk/CA/pdi-covid-19-ca-NegAntigen-Test.hl7" http://localhost:7071/api/reports
8. Verify there no QUALITY_FILTER NOT activated.
9. Run curl -X POST -H "client:prescryptive" -H "content-type:application/hl7-v2" --data-binary "@./junk/CA/pdi-covid-19-ca-PositiveAntigen-Test.hl7" http://localhost:7071/api/reports
10. Verify the QUALITY_FILTER activated.

Test files:

[pdi-covid-19-ca-NegAntigen-Test.txt](https://github.com/CDCgov/prime-reportstream/files/11511444/pdi-covid-19-ca-NegAntigen-Test.txt)
[pdi-covid-19-ca-PositiveAntigen-Test.txt](https://github.com/CDCgov/prime-reportstream/files/11511445/pdi-covid-19-ca-PositiveAntigen-Test.txt)



## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
